src/manifolds/ShapeSpace.jl

```diff
@@ -0,0 +1,241 @@
+
+@doc raw"""
+    KendallsPreShapeSpace{n,k} <: AbstractSphere{ℝ}
+
+Kendall's pre-shape space of ``k`` landmarks in $ℝ^n$ represented by n×k matrices.
+
+# Constructor 
+
+    KendallsPreShapeSpace(n::Int, k::Int)
+"""
+struct KendallsPreShapeSpace{n,k} <: AbstractSphere{ℝ} end
+
+KendallsPreShapeSpace(n::Int, k::Int) = KendallsPreShapeSpace{n,k}()
+
+function active_traits(f, ::KendallsPreShapeSpace, args...)
+    return merge_traits(IsEmbeddedSubmanifold())
+end
+
+representation_size(::KendallsPreShapeSpace{n,k}) where {n,k} = (n, k)
+
+"""
+    check_point(M::KendallsPreShapeSpace, p; atol=sqrt(max_eps(X, Y)), kwargs...)
+
+Check whether `p` is a valid point on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_point(M::KendallsPreShapeSpace, p; atol=sqrt(eps(eltype(p))), kwargs...)
+    for p_row in eachrow(p)
+        if !isapprox(mean(p_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(p_row),
+                "The point $(p) does not lie on the $(M) since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+"""
+    check_vector(M::KendallsPreShapeSpace, p, X; kwargs... )
+
+Check whether `X` is a valid tangent vector on [`KendallsPreShapeSpace`](@ref), i.e. whether
+each row has zero mean. Other conditions are checked via embedding in [`ArraySphere`](@ref).
+"""
+function check_vector(M::KendallsPreShapeSpace, p, X; atol=sqrt(max_eps(X, Y)), kwargs...)
+    for X_row in eachrow(X)
+        if !isapprox(mean(X_row), 0; atol, kwargs...)
+            return DomainError(
+                mean(X_row),
+                "The vector $(X) is not a tangent vector to $(p) on $(M), since one of the rows does not have zero mean.",
+            )
+        end
+    end
+    return nothing
+end
+
+embed(::KendallsPreShapeSpace, p) = p
+embed(::KendallsPreShapeSpace, p, X) = X
+
+function get_embedding(::KendallsPreShapeSpace{N,K}) where {N,K}
```

Comment:
Can we document this, that the pre shape space is embedded into its total space (of the quotient)?

Reply:
Done.